applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_linear_elastic_3D_law.cpp

```diff
@@ -0,0 +1,175 @@
+{
+    ConstitutiveLaw::Parameters parameters;
+    parameters.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR);
+    parameters.Set(ConstitutiveLaw::COMPUTE_STRESS);
+
+    auto strain = Vector{ScalarVector{6, 1.0}};
+    parameters.SetStrainVector(strain);
+
+    Vector stress;
+    parameters.SetStressVector(stress);
+
+    Matrix constitutive_matrix;
+    parameters.SetConstitutiveMatrix(constitutive_matrix);
+
+    Properties properties;
+    properties.SetValue(YOUNG_MODULUS, 1.0e7);
+    properties.SetValue(POISSON_RATIO, 0.3);
+    parameters.SetMaterialProperties(properties);
+
+    rConstitutiveLaw.CalculateMaterialResponsePK2(parameters);
+
+    return stress;
+}
+
+GeoIncrementalLinearElasticLaw CreateLinearElastic3DLaw()
+{
+    return GeoIncrementalLinearElasticLaw{std::make_unique<Elastic3D>()};
+}
+
+} // namespace
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElastic3DLawReturnsExpectedLawFeatures, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElastic3DLaw();
+
+    ConstitutiveLaw::Features law_features;
+    law.GetLawFeatures(law_features);
+
+    KRATOS_EXPECT_TRUE(law_features.mOptions.Is(ConstitutiveLaw::THREE_DIMENSIONAL_LAW))
+    KRATOS_EXPECT_TRUE(law_features.mOptions.Is(ConstitutiveLaw::INFINITESIMAL_STRAINS))
+    KRATOS_EXPECT_TRUE(law_features.mOptions.Is(ConstitutiveLaw::ISOTROPIC))
+
+    const auto& strain_measures = law_features.mStrainMeasures;
+    KRATOS_EXPECT_NE(std::find(strain_measures.begin(), strain_measures.end(), ConstitutiveLaw::StrainMeasure_Infinitesimal),
+                     strain_measures.end());
+    KRATOS_EXPECT_NE(std::find(strain_measures.begin(), strain_measures.end(),
+                               ConstitutiveLaw::StrainMeasure_Deformation_Gradient),
+                     strain_measures.end());
+
+    KRATOS_EXPECT_EQ(law_features.mStrainSize, 6);
+    KRATOS_EXPECT_EQ(law_features.mSpaceDimension, 3);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElastic3DLawReturnsExpectedStrainSize, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto law = CreateLinearElastic3DLaw();
+    KRATOS_EXPECT_EQ(law.GetStrainSize(), 6);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElastic3DLawReturnsExpectedWorkingSpaceDimension, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElastic3DLaw();
+    KRATOS_EXPECT_EQ(law.WorkingSpaceDimension(), 3);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElastic3DLawReturnsExpectedStress, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElastic3DLaw();
+
+    const auto stress = Calculate3DStress(law);
+
+    Vector expected_stress{6};
+    expected_stress <<= 2.5e+07, 2.5e+07, 2.5e+07, 3.84615e+06, 3.84615e+06, 3.84615e+06;
+    KRATOS_EXPECT_VECTOR_RELATIVE_NEAR(expected_stress, stress, 1e-3);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElastic3DLawReturnsExpectedStress_WhenOnlyDiagonalEntriesAreConsidered,
```

Comment:
Yes, really good to test this as it is a necessity for doing the K0 things in 3D.